.github/workflows/pkg-deps.yaml

```
@@ -0,0 +1,102 @@
+name: Package dependencies
+
+on:
+  workflow_call:
+
+jobs:
+  check-dependency:
+    name: Check dependency
+    runs-on: ubuntu-latest
+    if: |
+      github.event_name == 'pull_request' &&
+      startswith(github.base_ref, 'ubuntu-')
+    env:
+      branch: ${{ github.base_ref }}
+      msg_file: dependencies.message
+    permissions:
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check changed paths
+        id: changed-paths
+        uses: dorny/paths-filter@v3
+        with:
+          # ref: https://github.com/marketplace/actions/paths-changes-filter
+          filters: |
+            slices:
+              - added|modified: 'slices/**/*.yaml'
+          # Space delimited list usable as command-line argument list in
+          # Linux shell. If needed, it uses single or double quotes to
+          # wrap filename with unsafe characters.
+          list-files: shell
+
+      - name: Check dependencies
+        id: check-deps
+        if: steps.changed-paths.outputs.slices == 'true'
+        run: |
+          set -e
+
+          export LC_COLLATE=C
+
+          version=$(echo "$branch" | grep -Eo '[0-9.]+')
+          docker run -i -d --rm --name ubuntu ubuntu:"$version"
+          docker exec ubuntu apt-get update
+
+          cleanup() {
+            docker stop ubuntu
+          }
+          trap cleanup EXIT
+
+          echo "Slice package dependency issues:" > "$msg_file"
```

```suggestion
          echo "Diff of dependencies:" > "$msg_file"
```

let's not call it "issues". We don't want to convey the message that something is wrong. It might not be. this is purely informative.